common/system/system.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright IBM Corp All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package system
+
+import (
+	"context"
+	"crypto/tls"
+	"net"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/hyperledger/fabric/common/flogging"
+	"github.com/hyperledger/fabric/common/util"
+	"github.com/hyperledger/fabric/core/middleware"
+)
+
+//go:generate counterfeiter -o fakes/logger.go -fake-name Logger . Logger
+
+type Logger interface {
+	Warn(args ...interface{})
+	Warnf(template string, args ...interface{})
+}
+
+type Options struct {
+	Logger        Logger
+	ListenAddress string
+	TLS           TLS
+}
+
+type System struct {
```

Comment:
This name stutters, and I'm generally a little confused by the choice of package name.  As a reader I would struggle to know what a `system.System` logically represents.  I'd invite feedback for a better name... the more natural one would be `http.Server` or similar, but there is the obvious mental conflict with the stdlib.  Maybe something like `fabhttp.Server` or `mtlshttp.Server` or... hopefully someone else has a better thought, but I really don't love `system.System`.

Reply:
Agreed. Of those options, fabhttp.Server seems more obvious to me in terms of describing what it is.

Reply:
Renamed to fabhttp.Server for now.